### ISSUE TITLE
IPv6-nat container keeps restarting

### ISSUE THREAD

**Issue by SomeGeek** [mailcow/mailcow-dockerized#2404]
When disabling docker's iptables, the ipv6nat container keeps restarting. I would like to manage the firewall rules using iptables myself, because docker's generated iptables rules conflicts other software on the host.

Can I safely ignore this? Or is there a better workaround?

**Comment by andryyy** [comment]
You can remove the container from the compose file or add a sleep command to the override yml for this service.

**Comment by SomeGeek** [comment]
Great. It would be nice if there would be some documentation about manual IPv6 natting, without the ipv6nat container. 

Is it correct that it comes down to creating DNAT-rules for the mailservices? If you can provide me a list, will try to write some documentation about the subject when I have time.

**Comment by andryyy** [comment]
No, sorry.

**Comment by SomeGeek** [comment]
For those of you who like to read how it turned out for me, read on. I decided to use nftables instead of iptables, since it will be the new 'standard' (Debian Buster and RHEL 8 are going to use it as default, for example).

I had it working, until I realized, after starting mailcow, the iptables modules were added to the kernel (even when I disabled the modules). This led to a situation I want to avoid: I only wanted to use nftables, nothing else. Having both running at the same time leads to a flaky situation. I have not tried to blacklist them, but as you read next, this still would cause a unwanted situation.

I found out that fail2ban uses a hardcoded iptables implementation. Using nftables would mean giving up fail2ban protection, which is a big drawback for me (especially on a mailserver). I tried to come up with a solution for this, which meant modifying mailcow's code (which I try to avoid).

So I went for a different solution and installed mailcow on it's own server instead and doing iptables manually... which worked... for IPv4. Unfortunately, docker's native IPv6 support is rubbish. It doesn't even add a DOCKER-USER chain to ip6tables.  This is also why mailcow uses this ipv6-nat container in the fist place. So I felt I was back where I started, sigh...

The only workaround at this point is loading my own chains directly after the MAILCOW chain in ip6tables (which mimmicks the DOCKER-USER chain behavior on iptables), but that can only be applied _after_ mailcow has been loaded, where AFAIK Docker creates the DOCKER-USER chain when it starts. There are some issues open in the docker repository's with recent activity, yet they don't feel the need to prioritize it. At this point in time, the whole IPv6 thing in Docker feels like a terrible hack to me.

The best solution I can think of is making the netfilter container compatible with nftables. This would make fail2ban work the way it should be and giving anyone the freedom to make his own custom firewall ruleset.
